mlir/include/mlir/Dialect/Vector/Transforms/VectorRewritePatterns.h

```diff
@@ -147,6 +147,23 @@ void populateVectorTransferCollapseInnerMostContiguousDimsPatterns(
 void populateSinkVectorBroadcastPatterns(RewritePatternSet &patterns,
                                          PatternBenefit benefit = 1);
 
+/// Patterns that fold chained vector reductions. These patterns assume that
+/// vector addition (e.g., `arith.addf` with vector operands) is cheaper than
```

Comment:
This pattern is not meant to restrict to addition only? Maybe "vector elementwise operations"?

Reply:
Clarified. In principle yes, but currently only addition is handled